test/unittest/blas3/gemm/blas3_gemm_def.hpp

```diff
@@ -49,46 +51,171 @@ TYPED_TEST(BLAS_Test, gemm) {
   const char* ta_str = MatAType::str;
   const char* tb_str = MatBType::str;
 
-  std::array<size_t, 2> dim_a = {127, 127};
-  std::array<size_t, 2> dim_b = {127, 127};
-  std::array<size_t, 2> dim_c = {127, 127};
-
+  auto _TransA = tolower(*ta_str);
+  auto _TransB = tolower(*tb_str);
+  bool _TrA = _TransA != 'n';
+  bool _TrB = _TransB != 'n';
   ScalarT alpha = ScalarT(1);
   ScalarT beta = ScalarT(1);
-
-  std::vector<ScalarT> a_m(dim_a[0] * dim_a[1]);
-  std::vector<ScalarT> b_m(dim_b[0] * dim_b[1]);
-  std::vector<ScalarT> c_m_gpu_result(dim_c[0] * dim_c[1], ScalarT(0));
-  std::vector<ScalarT> c_m_cpu(dim_c[0] * dim_c[1], ScalarT(0));
-  TestClass::set_rand(a_m, dim_a[0] * dim_a[1]);
-  TestClass::set_rand(b_m, dim_b[0] * dim_b[1]);
-  auto lda = dim_a[0];
-  auto ldb = dim_b[0];
-  auto ldc = dim_c[0];
-  auto m = dim_c[0];
-  auto n = dim_c[1];
-  auto k = dim_a[1];
-  gemm(ta_str, tb_str, m, n, k, alpha, a_m.data(), lda, b_m.data(), ldb, beta,
-       c_m_cpu.data(), m);
   SYCL_DEVICE_SELECTOR d;
   auto q = TestClass::make_queue(d);
   Executor<ExecutorType> ex(q);
-  auto m_a_gpu = ex.template allocate<ScalarT>(dim_a[0] * dim_a[1]);
-  auto m_b_gpu = ex.template allocate<ScalarT>(dim_b[0] * dim_b[1]);
-  auto m_c_gpu = ex.template allocate<ScalarT>(dim_c[0] * dim_c[1]);
-  ex.copy_to_device(a_m.data(), m_a_gpu, dim_a[0] * dim_a[1]);
-  ex.copy_to_device(b_m.data(), m_b_gpu, dim_b[0] * dim_b[1]);
-  ex.copy_to_device(c_m_gpu_result.data(), m_c_gpu, dim_c[0] * dim_c[1]);
-  _gemm(ex, *ta_str, *tb_str, m, n, k, alpha, m_a_gpu, lda, m_b_gpu, ldb, beta,
-        m_c_gpu, ldc);
-  auto event =
-      ex.copy_to_host(m_c_gpu, c_m_gpu_result.data(), dim_c[0] * dim_c[1]);
-  ex.wait(event);
-
-  for (size_t i = 0; i < dim_c[0] * dim_c[1]; ++i) {
-    ASSERT_NEAR(c_m_gpu_result[i], c_m_cpu[i], prec);
+  int batch_sizes[] = {1,
+                       5
+#ifdef AGGRESSIVE_TEST
+                       ,
+                       11,
+                       2,
+                       13,
+                       31,
+                       39,
+                       63,
+                       64,
+                       65,
+                       127,
+                       129,
+                       255,
+                       257,
+                       511,
+                       512,
+                       513,
+                       1023,
+                       1024,
+                       1025
+#endif
+
+  };
+
+  int m_sizes[] = {11,
+                   33,
+                   65,
+                   129,
+                   255,
+                   513
+#ifdef AGGRESSIVE_TEST
+                   ,
+                   2,
+                   14,
+                   39,
+                   63,
+                   64,
+                   127,
+                   257,
+                   511,
+                   512,
+                   1023,
+                   1024,
+                   1025
+#endif
```

Comment:
This will change when we restructure our test framework. So in this case the range will be pass as an input at runtime. The ticket for it has been created

Reply:
I have replaced them with std::array